src/GitVersionCore/Configuration/ConfigurationBuilder.cs

```diff
@@ -60,7 +60,7 @@ private static void ApplyOverrides([NotNull] Config targetConfig, [NotNull] Conf
             targetConfig.CommitMessageIncrementing = overrideConfig.CommitMessageIncrementing ?? targetConfig.CommitMessageIncrementing;
             targetConfig.Increment = overrideConfig.Increment ?? targetConfig.Increment;
             targetConfig.CommitDateFormat = overrideConfig.CommitDateFormat ?? targetConfig.CommitDateFormat;
-            targetConfig.MergeMessageFormats = overrideConfig.MergeMessageFormats ?? targetConfig.MergeMessageFormats;
+            targetConfig.MergeMessageFormats = overrideConfig.MergeMessageFormats.Any() ? overrideConfig.MergeMessageFormats : targetConfig.MergeMessageFormats;
```

Comment:
Is there any chance of `MergeMessageFormats` being `null`, which can cause a `NullReferenceException` on the call to `Any()`?

```suggestion
            targetConfig.MergeMessageFormats = overrideConfig.MergeMessageFormats?.Any() ? overrideConfig.MergeMessageFormats : targetConfig.MergeMessageFormats;
```

Reply:
The default value is a Dictionary actually.
``
        [YamlMember(Alias = "merge-message-formats")]
        public Dictionary<string, string> MergeMessageFormats { get; set; } = new Dictionary<string, string>();
``
So as far as I know it should be okay.